_posts/2020-05-22-smallrye-config-1-8.adoc

```diff
@@ -0,0 +1,153 @@
+= SmallRye Config 1.8 Released
+:page-layout: post
+:page-title: SmallRye Config 1.8 Released
+:page-synopsis: SmallRye Config 1.8 Released
+:page-tags: [microprofile,smallrye,config]
+:page-date: 2020-05-22 13:00:00.000 +0100
+:page-author: radcortez
+
+SmallRye Config version 1.8 was released a few days ago.
+
+The latest version of SmallRye Config introduced a few experimental changes to enhance the API and to make SmallRye
+Config more extensible. One of the experimental changes is the new Interceptor model, to intercept the lookup of a
+configuration value.
+
+== The ConfigSourceInterceptor
+
+The `ConfigSourceInterceptor` allows to intercept the resolution of a configuration name before the `Config` resolves
+the configuration value and before any conversion taking place. This is useful to implement features like
+Property Expansion or Configuration Profiles.
+
+Implementations of `ConfigSourceInterceptor` are loaded via the `java.util.ServiceLoader` mechanism and can be
+registered by providing a resource named `META-INF/services/io.smallrye.config.ConfigSourceInterceptor`, which
+contains the fully qualified `ConfigSourceInterceptor` implementation class name as its content.
+
+A single method must be implemented, `ConfigValue getValue(ConfigSourceInterceptorContext context, String name)`. The
+`ConfigSourceInterceptorContext` is used to proceed with the interceptor chain. The chain can be short-circuited by
+returning an instance of `ConfigValue`.
+
+[source,java]
+----
+public interface ConfigSourceInterceptor extends Serializable {
+    ConfigValue getValue(ConfigSourceInterceptorContext context, String name);
+}
+----
+
+The `ConfigValue` objects hold information about the key name, value, config source origin, and ordinal.
+
+Additionally, the `ConfigSourceInterceptor` may also intercept resolution of configuration names of configuration
+`ConfigValue` values with the methods `Iterator<String> iterateNames(ConfigSourceInterceptorContext context)` and
```

Comment:
"resolution of configuration names of configuration `ConfigValue` values" is hard to follow. Would "resolution of configuration names" be enough?

Actually looking at what follows, it seems it's mainly a typo -- "of" instead of "or". I'd still get rid of the "`ConfigValue`" word -- just "resolution of configuration names or configuration values".

Reply:
Yeah it was a typo :)